fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(res => {
+        this.setState({
+          todos: res
+        });
+      });
+  }
+
+  addTodo(newTodo) {
+    const postOptions = {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        id: newTodo.id,
+        task: newTodo.task,
+        isCompleted: newTodo.isCompleted
+      })
+    };
+    fetch('/api/todos', postOptions)
+      .then(newTodo => newTodo.json())
+      .then(newTodo => {
+        this.state.todos.push(newTodo);
+        this.getAllTodos();
```

Comment:
`.push()` modifies the original array, but we need to create a new array before the push. Also, you have the old list in state, and the new todo from the server, no need to re-fetch everything. we update state with `.setState()` in react